toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -185,13 +313,206 @@ func resolveGraphNodes(dependencyGraph *pkggraph.PkgGraph, inputSummaryFile, out
+				// the delta RPM since it doesn't exist yet.
+				skippedNodes = append(skippedNodes, n)
+			}
+		}
+	}
 
-	if strings.TrimSpace(outputSummaryFile) != "" {
-		err = repoutils.SaveClonedRepoContents(cloner, outputSummaryFile)
+	// Fix up the implicit nodes to point to the correct delta RPMs that we parsed earlier.
+	for _, n := range skippedNodes {
+		logger.Log.Debugf("Fixing up skipped node %s", n)
+		err = fixupDeltaImplicitNode(realDependencyGraph, n, &originalPathToDeltaPathMap)
 		if err != nil {
-			logger.Log.Errorf("Failed to save cloned repo contents.")
-			return
+			return fmt.Errorf("failed to fixup skipped node %s: %w", n, err)
+		}
+	}
+
+	logger.Log.Info("Configuring additional delta RPMs as a local repository")
+	err = cloner.ConvertDownloadedPackagesIntoRepo()
+	if err != nil {
+		logger.Log.Errorf("Failed to convert downloaded RPMs into a repo. Error: %s", err)
+		return
+	}
+
+	return
+}
+
+// downloadSingleDeltaRPM attempts to download a single delta RPM for a build node. If the delta RPM is available
+// it will be downloaded and the build node will be updated to point to the new RPM. The associated run node will
+// also be updated to point to the new RPM since the scheduler uses the run node to find the RPM to install.
+//   - realDependencyGraph: The graph to update
+//   - realBuildNode: The build node to update. This node should be from the real graph as we will be updating it directly.
+//     to find the actual build node in the graph.
+//   - cloner: The cloner to use to download the RPMs
+//   - deltaRpmDir: The directory to download the RPMs into (likely the same as the normal RPM cache)
+//   - pathMap: A map of the original path to the delta path for each RPM that was downloaded. This is used to fix up
+//     the implicit nodes later.
+func downloadSingleDeltaRPM(realDependencyGraph *pkggraph.PkgGraph, realBuildNode *pkggraph.PkgNode, cloner *rpmrepocloner.RpmRepoCloner, deltaRpmDir string, pathMap *map[string]string) (foundMatch bool, err error) {
+	//TODO: downloadDependencies is broken right now, fix it
+	const downloadDependencies = true
+	var lookup *pkggraph.LookupNode
+
+	// Find the real build node in the graph we want to keep (we will be discarding the graph the node was passed in from so we can't use it)
+	if realBuildNode.Type != pkggraph.TypeBuild {
+		err = fmt.Errorf("node '%s' is not a build node, can't download delta RPM", realBuildNode)
+		return false, err
+	}
+
+	timestamp.StartEvent(fmt.Sprintf("downloading delta node %s", realBuildNode.VersionedPkg.Name), nil)
+	defer timestamp.StopEvent(nil)
+
+	lookup, err = realDependencyGraph.FindExactPkgNodeFromPkg(realBuildNode.VersionedPkg)
+	if err != nil {
+		err = fmt.Errorf("can't find build node '%s' in graph: %w", realBuildNode, err)
+		return false, err
+	}
+	if lookup == nil || lookup.RunNode == nil {
+		err = fmt.Errorf("can't find run lookup '%v' in graph", lookup)
+		return false, err
+	}
+
+	if lookup.BuildNode != realBuildNode {
```

Comment:
>lookup.BuildNode != realBuildNode [](http://example.com/codeflow?start=4&length=33)

Is it even possible to get a different node from `FindExactPkgNodeFromPkg`? We shouldn't need to double-guess our own functions. If they produce results, which shouldn't happen, then that sounds like a bug we should be fixing.

Reply:
This one is fair, if this goes wrong we will probably not have made it this far before something exploded anyways.